spec/Appendix A -- Field Selection.md

```diff
@@ -0,0 +1,797 @@
+This path is defined as a sequence of field names, each separated by a period (`.`) to create segments. 
+
+``` example
+book.title
+```
+
+Each segment specifies a field in the context of the parent, with the root segment referencing a field in the _return type_ of the query. 
+Arguments are not allowed in a {Path}.
+
+To select a field when dealing with abstract types, the segment selecting the parent field must specify the concrete type of the field using angle brackets after the field name if the field is not defined on an interface.
+
+In the following example, the path `mediaById<Book>.isbn` specifies that `mediaById` returns a `Book`, and the `isbn` field is selected from that `Book`.
+
+``` example
+mediaById<Book>.isbn
+```
+
+### SelectedValue
+SelectedValue :: 
+    - Path
+    - SelectedObjectValue
+    - Path . SelectedObjectValue
+    - SelectedValue | SelectedValue 
+
+A {SelectedValue} is defined as either a {Path} or a {SelectedObjectValue}
+
+A {Path} is designed to point to only a single value, although it may reference multiple fields depending on the return type.
+To allow selection from different paths based on type, a {Path} can include multiple paths separated by a pipe (`|`).
+
+In the following example, the value could be `title` when referring to a `Book` and `movieTitle` when referring to a `Movie`. 
+
+``` example
+mediaById<Book>.title | mediaById<Movie>.movieTitle
+```
+
+The `|` operator can be used to match multiple possible {SelectedValue}.
+This operator is applied when mapping an abstract output type to a `@oneOf` input type.
+
+```example
+{ movieId: <Movie>.id } | { productId: <Product>.id }
+```
+
+```example
+{ nested: { movieId: <Movie>.id } | { productId: <Product>.id }}
+```
+
+### SelectedObjectValue
+SelectedObjectValue :: 
+    - { SelectedObjectField+ }
+
+SelectedObjectField :: 
+    - Name: SelectedValue
+
+{SelectedObjectValue} are unordered lists of keyed input values wrapped in curly-braces `{}`.
+It has to be used when the expected input type is an object type.
+
+This structure is similar to the `ObjectValue` defined in the GraphQL specification, but it differs by allowing the inclusion of {Path} values within a {SelectedValue}, thus extending the traditional `ObjectValue` capabilities to support direct path selections.
+
+A {SelectedObjectValue} following a {Path} is scoped to the type of the field selected by the {Path}.
+This means that the root of all {SelectedValue} inside the selection is no longer scoped to the root (defined by `@is` or `@require`) but to the field selected by the {Path}. The {Path} does not effect the structure of the input type.
+
+This allows to reduce repetition in the selection.
+
+The following example is valid:
+
+```graphql example
+type Product {
+    dimension: Dimension!
+    shippingCost(dimension: DimensionInput! @require(field: "dimension.{ size weight }")): Int! @lookup
```

Comment:
i think those `@lookup`s should be dropped?

Reply:
Done